src/frontends/lean/scanner.cpp

```diff
@@ -704,7 +704,7 @@ void scanner::skip_to_pos(pos_info const & pos) {
     for (unsigned line_no = 1; line_no < pos.first; line_no++)
         fetch_line();
     m_line = m_sline;
-    for (unsigned col_idx = 0; col_idx < pos.second; col_idx++)
+    while (static_cast<unsigned>(m_upos) < pos.second)
```

Comment:
@gebner This triggered the assertion below, but curiously only when navigating to the last line of a file. Couldn't reproduce it in a test case either.